src/lib/y2partitioner/widgets/btrfs_subvolumes.rb

```diff
@@ -0,0 +1,82 @@
+require "yast"
+require "y2partitioner/widgets/btrfs_subvolumes_table"
+require "y2partitioner/widgets/btrfs_subvolumes_add_button"
+require "y2partitioner/widgets/btrfs_subvolumes_delete_button"
+require "y2partitioner/device_graphs"
+
+Yast.import "Mode"
+
+module Y2Partitioner
+  module Widgets
+    # Widget to manage btrfs subvolumes of a specific filesystem
+    #
+    # FIXME: How to handle events directly from a CWM::Dialog ?
+    # Events for :help and :cancel buttons should be managed from the dialog,
+    # for example to show a popup with the help.
+    class BtrfsSubvolumes < CWM::CustomWidget
+      attr_reader :filesystem
+
+      # @param filesystem [Y2Storage::Filesystems::BlkFilesystem] a btrfs filesystem
+      def initialize(filesystem)
+        @filesystem = filesystem
+
+        self.handle_all_events = true
+      end
+
+      # FIXME: The help handle does not work without wizard
+      #
+      # This handle should belongs to the dialog
+      # @see Dialogs::BtrfsSubvolumes
+      def handle(event)
+        handle_help if event["ID"] == :help
+        nil
+      end
+
+      def contents
+        table = Widgets::BtrfsSubvolumesTable.new(filesystem)
+
+        VBox(
+          # TODO: libstorage must manage snapshots
+          HBox(HSpacing(1), Left(CheckBox(_("Enable snapshots"), false))),
+          VSpacing(1),
+          table,
+          HBox(
+            Widgets::BtrfsSubvolumesAddButton.new(table),
+            Widgets::BtrfsSubvolumesDeleteButton.new(table)
+          )
+        )
+      end
+
+      # FIXME: this should belongs to the dialog
```

Comment:
well, in fact this help is fine to have here. PRoblem I see is `handle_help` which is dialog job to collect help. ( so move FIXME to that handle method ).